js_modules/dagster-ui/packages/ui-core/src/graph/KindTags.tsx

```diff
@@ -62,17 +59,14 @@ export const AssetComputeKindTag = ({
         style={{...style, cursor: shouldLink || currentPageFilter ? 'pointer' : 'default'}}
         tags={[
           {
-            label: definition.computeKind,
-            onClick:
-              currentPageFilter && definition.computeKind
-                ? () => currentPageFilter.setState(new Set([definition.computeKind || '']))
-                : shouldLink
-                ? () => {
-                    window.location.href = linkToAssetTableWithComputeKindFilter(
-                      definition.computeKind || '',
-                    );
-                  }
-                : () => {},
+            label: computeKind,
+            onClick: currentPageFilter
+              ? () => currentPageFilter.setState(new Set([computeKind || '']))
+              : shouldLink
+              ? () => {
+                  window.location.href = linkToAssetTableWithComputeKindFilter(computeKind || '');
```

Comment:
use `history.pushState` instead of assigning to `window.location.href` directly.

Reply:
This is existing code that's just being refactored. I would prefer to make your recommended change in a separate PR

Reply:
Understood that it's existing code but might as well improve it while we're changing it and it's just a 2 line change to use `const history = useHistory()` `history.push(...)`

Reply:
There were other places that it was happening so put it in https://github.com/dagster-io/dagster/pull/22713